napari/_qt/qt_viewer.py

```diff
@@ -1034,6 +1062,27 @@ def toggle_console_visibility(self, event=None):
             self.viewerButtons.consoleButton
         )
 
+    def on_fps_update(self, event):
+        if not self._camera_monitor.camera_moving:
+            return
+        fps = event.fps
+        if fps < self.viewer.target_fps[0]:
+            logging.info('decrease render quality')
+            for layer in self.viewer.layers:
+                layer.change_render_quality(RenderQualityChange.DECREASE)
+        if fps > self.viewer.target_fps[1]:
+            logging.info('increase render quality')
+            for layer in self.viewer.layers:
+                layer.change_render_quality(RenderQualityChange.INCREASE)
+
+    @qdebounced(timeout=1500, leading=False)
```

Comment:
```suggestion
    @qdebounced(timeout=1500, leading=True)
```
Using True here makes the higher quality refresh at the end snappier.

Reply:
... That doesn't make sense, are you sure about it? We can change the timeout but we really do want to wait until all the camera move events are done. See:

https://css-tricks.com/debouncing-throttling-explained-examples/#aa-leading-edge-or-immediate

Using leading=True should trigger a high-resolution draw at the *start* of moving a camera, which is not what you want...

Reply:
It makes a huge difference for me.

I think the behavior is like your doc link (cool stuff btw).
The callback is called when mouse move *finishes*, but with leading = false, the debouncer waits for 1.5s before triggering the callback -- even if just 1 event fired. So we get the delay of the debounce plus the slow final render at 0.1.
Meanwhile, with leading = true on the debouncer, the callback will fire so as soon as the event triggers (mouse move stops) -- but then won't retrigger unless the denounce time passes.

Honestly, I'm not even sure the `moving_finished` callback needs to be debounced? It's pretty hard to start-n-stop dragging in 1.5 s, and even then wouldn't you want the final render to fire anyways?

Or is the idea here of someone who pauses for a moment, so we delay the final render?
If that's the case, could just drop the debounce and step down the step size, rather than doing it in one pass. It would still be more responsive than the present.

Reply:
The idea with the debounce is that you might want to click-drag, then click-drag again because you haven't got to where you've got. ie repeated rotates or repeated pans. It would be annoying if in that gap you triggered this super expensive draw and now your second drag is jerky again.

I do think 1.5s is too conservative for this. But I don't think 0 debounce is the answer... (Well, have you tried it? Ideally with a really taxing dataset, something for which you'd want the 80 step size.)

Reply:
I still think when I stop moving, I want it to render otherwise hard to say if i'm where I want to be. 
I've removed the debounce entirely. The laggy-ness is really annoying.

Reply:
@psobolewskiPhD ooooooh, so I just removed it, and was puzzled when ruff didn't complain that I had an unused qdebounced import. It turns out that the camera monitor "stopped moving" signal is itself debounced 500ms, after which we were waiting another 1.5s for the callback. All the mysteries are solved. 😅 

Just to clarify, you didn't remove the debounced on the camera monitor, did you?

Reply:
correct, just the 1.5s one